redash/query_runner/mixpanel.py

```diff
@@ -0,0 +1,198 @@
+        else:
+            Mixpanel.LOGGER.warning("Maximum retries reached. Request failed. Try again later.")
+            raise BaseException
+
+
+def flatten_row(row):
+    result = {}
+    for k in row:
+        if type(row[k]) is dict:
+            for j in row[k]:
+                result[k + "_" + j] = row[k][j]
+        elif type(row[k]) is list:
+            for index, j in enumerate(row[k]):
+                result[k + "_" + str(index)] = row[k][index]
+        else:
+            result[k] = row[k]
+    return result
+
+
+class MixpanelJql(BaseQueryRunner):
+    noop_query = '''
+        main = () => People().reduce(mixpanel.reducer.null())
+    '''
+
+    @classmethod
+    def annotate_query(cls):
+        return False
+
+    @classmethod
+    def type(cls):
+        return "mixpanel_jql"
+
+    @classmethod
+    def name(cls):
+        return "Mixpanel JQL"
+
+    def __init__(self, configuration):
+        super(MixpanelJql, self).__init__(configuration)
+
+    @classmethod
+    def configuration_schema(cls):
+        return {
+            'type': 'object',
+            'properties': {
+                'api_secret': {
+                    'type': 'string',
+                    'title': 'Api Secret'
+                },
+                'api_token': {
+                    'type': 'string',
+                    'title': 'Api Token'
+                }
+            },
+            "required": ["api_secret", "api_token"],
+            "secret": []
+        }
+
+    @classmethod
+    def enabled(cls):
+        return enabled
+
+    def run_query(self, query, user):
+        api_secret = self.configuration.get('api_secret')
+        api_token = self.configuration.get('api_token')
+        api = MixpanelPatched(api_secret, token=api_token)
+
+        try:
+            data = api.query_jql(query)
+        except urllib2.HTTPError as e:
+            if hasattr(e, 'read'):
+                error = json.loads(e.read())['error']
+            else:
+                error = str(e.reason)
+            return None, error
+        except Exception as e:
+            return None, e
+        else:
+            result = {}
+            if (not isinstance(data[0], dict)) or type(data[0]) in [str, unicode]:
+                result["columns"] = [{"name": "value", "friendly_name": "value"}]
+                result["rows"] = [{"value": v} for v in data]
+            elif isinstance(data[0], dict):
+                result["columns"] = []
+                for k in data[0]:
+                    if type(data[0][k]) is dict:
+                        for j in data[0][k]:
+                            result["columns"].append({"name": k + "_" + j, "friendly_name": k + "_" + j})
+                    elif type(data[0][k]) is list:
+                        for index, j in enumerate(data[0][k]):
+                            result["columns"].append({"name": k + "_" + str(index), "friendly_name": k + "_" + str(index)})
```

Comment:
Column also have a type property. You can use `redash.query_runner.guess_type` to "guess it" or at least assign `redash.query_runner.TYPE_STRING`.